### PR TITLE
Add SHA-256 and MD5 MessageDigest to CRIUSecProvider

### PR DESCRIPTION
**Pull request by WilburZjh**
The purpose of this PR is to expand the
security services within CRIUSecProvider
to also include SHA-256 and MD5 related
as per request.

Back-ported from https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/685

### CODE REVIEW

**Comment by WilburZjh** [comment]
Back-ported from https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/685

**Comment by WilburZjh** [comment]
@jasonkatonica FYI

**Comment by WilburZjh** [comment]
@keithc-ca Please help to review

File: closed/src/java.base/share/classes/openj9/internal/criu/CRIUSECProvider.java [hunk 1]
```diff
@@ -39,26 +42,46 @@ public final class CRIUSECProvider extends Provider {
...
 
         // SHA1PRNG is the default name needed by the jdk, but SHA1 is not used, rather it reads directly from /dev/random.
-        putService(new Service(this, "MessageDigest", "SHA-1", packageName + "SHA", java.util.Arrays.asList(aliases), null));
-        putService(new Service(this, "SecureRandom", "SHA1PRNG", packageName + "NativePRNG", null, null));
-    }
+        putService(new Service(this, "MessageDigest", "SHA-1", "sun.security.provider.SHA", java.util.Arrays.asList(aliases), null));
+        putService(new Service(this, "MessageDigest", "SHA-256", "sun.security.provider.SHA2$SHA256", null, null));
+        putService(new Service(this, "MessageDigest", "MD5", "sun.security.provider.MD5", null, null));
+        putService(new Service(this, "Mac", "HmacSHA256", "com.sun.crypto.provider.HmacCore$HmacSHA256", null, null));
+        putService(new Service(this, "SecureRandom", "SHA1PRNG", "sun.security.provider.NativePRNG$CRIUNativePRNG", null, null));    }
```
Comment (keithc-ca) [inline closed/src/java.base/share/classes/openj9/internal/criu/CRIUSECProvider.java:71]: Closing brace on the next line, please (as it was before).

File: src/java.base/unix/classes/sun/security/provider/NativePRNG.java [hunk 2]
```diff
@@ -565,9 +665,18 @@ private void implNextBytes(byte[] data) {
...
                 }
         }
+
+        /*[IF CRIU_SUPPORT]*/
+        private void clearRNGState() {
+            if (mixRandom != null) {
+                mixRandom.clearState();
+            }
+        }
+        /*[ENDIF] CRIU_SUPPORT */
```
Comment (keithc-ca) [inline src/java.base/unix/classes/sun/security/provider/NativePRNG.java:680]: The head stream has a blank line after this; there's no need for this file to be different.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/GensrcJ9JCL.gmk [hunk 3]
```diff
@@ -39,6 +39,7 @@ $(eval $(call SetupCopyFiles,COPY_OVERLAY_FILES, \
 	SRC := $(TOPDIR), \
 	DEST := $(SUPPORT_OUTPUTDIR)/overlay, \
 	FILES := \
+	    src/java.base/share/classes/com/sun/crypto/provider/HmacCore.java \
```
[the remark this answers is not in the document]
Comment (keithc-ca) [reply closed/GensrcJ9JCL.gmk]: The following lines are indented with tabs. Please do the same here.

[the remark this answers is not in the document]
Comment (WilburZjh) [reply closed/GensrcJ9JCL.gmk]: updated

[the remark this answers is not in the document]
Comment (WilburZjh) [reply closed/GensrcJ9JCL.gmk]: should be ok now.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**